### DISCUSSION TITLE
Eternal orchestrations and replay behavior - clarification

### DISCUSSION THREAD

**Discussion by obesser** [Azure/azure-functions-durable-extension#2461] [Q&A] [answered]
I'm following a sample of the monitoring pattern here:
https://github.com/Azure/azure-functions-durable-python/blob/dev/samples/aml_monitoring/aml_durable_orchestrator/__init__.py
where one activity function is starting a long external job and a second activity function polls its status.

A while loop is used to for the polling activity (line 20), but there is also a comment about using continueAsNew instead (line 18).

How is it possible to implement the eternal pattern in this case? To my understanding, with continueAsNew, on each execution, the activity functions will run again and not replay, meaning a new monitoring instance each time..
Is it possible, after a continueAsNew, to only replay the result of the first activity (start job) but execute the second activity (status polling)?

**Comment by cgillum** [comment, marked as answer]
The way to accomplish this is to have some flag in your orchestration input that tells you whether you've executed the first activity (the start job). In the initial invocation of your orchestration, this flag should be set to false, ensuring that the first activity runs. As part of the first `continueAsNew` call, you'd set this flag to true so that all future invocations don't call this first activity again.

**Reply by obesser** [reply]
Thanks @cgillum !
